src/agent.rs

```diff
@@ -189,27 +189,16 @@ impl Agent {
         Ok(())
     }
 
-    /// Receive content sent from the other side and stored in the return value
+    /// Receive content sent from the other side and stored in the `LocalMr`
     pub(crate) async fn receive_data(&self) -> io::Result<LocalMr> {
-        let data = self
+        let (lmr, len) = self
             .data_recv
             .lock()
             .await
             .recv()
             .await
             .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "data channel closed"))?;
-        // TODO: avoid data copy or provide another api which is quick but ugly.
-        let mut res_lmr = self
-            .inner
-            .allocator
-            // alignment 1 is always correct
-            .alloc(unsafe { &Layout::from_size_align_unchecked(data.1, 1) })?;
-        res_lmr.as_mut_slice().copy_from_slice(
-            data.0
-                .get(*SEND_DATA_OFFSET..(SEND_DATA_OFFSET.overflow_add(data.1)))?
-                .as_slice(),
-        );
-        Ok(res_lmr)
+        lmr.take(0..len)
```

Comment:
Will the length of original lmr be lost?

Reply:
yes, and we don't need users know the meaningless content in range [len..] of the original lmr. So I think lost the original length is ok.

Reply:
Local mr is just a wrapper of raw mr for now, and the length of raw mr won't be lost, so losing lmr's length does not cause a memory leak.

Reply:
Does it effect the LocalMr release operation? Especially the allocator function.

Reply:
No, when LocalMr release, the related memory space will be released according to the `addr` and `len` of RawMr which referenced by the releasing LocalMr.